dGame/dGameMessages/GameMessages.cpp

```diff
@@ -4733,12 +4733,17 @@ void GameMessages::HandleBuyFromVendor(RakNet::BitStream* inStream, Entity* enti
 
 	const auto isCommendationVendor = entity->GetLOT() == 13806;
 
-	VendorComponent* vend = static_cast<VendorComponent*>(entity->GetComponent(eReplicaComponentType::VENDOR));
+	auto* vend = entity->GetComponent<VendorComponent>();
 	if (!vend && !isCommendationVendor) return;
 
-	InventoryComponent* inv = static_cast<InventoryComponent*>(player->GetComponent(eReplicaComponentType::INVENTORY));
+	auto* inv = player->GetComponent<InventoryComponent>();
 	if (!inv) return;
 
+	if (!isCommendationVendor && !vend->SellsItem(item)) {
```

Comment:
Non-blocking
Should there be a check for the commendation vendor as well? shouldn't be toooooo hard :P
But should probably be it's own PR since it will be a bit more involved

Reply:
He has his own check below.  It checks that the item is actually sold for tokens hence dont check here